### PR TITLE
Feature/step one

### PR DESCRIPTION
**Pull request by arnoriel**
# Pull Request

## Issue(s)

Error Code Responses still Missleading

## Solution(s)

Changing Error Code responses

## App Configuration Change(s)

nothing to do

## Documentation(s)/Reference(s) (if any)

- Register error with empty username:
Screenshot (470)
- Register error with empty email:
Screenshot (469)
- Register error with empty Password 1:
Screenshot (471)
- Register error with empty Password 2:
Screenshot (472)
- Register error with unmatched Password:
Screenshot (467)
- Register Success:
Screenshot (473)
- Get all Users:
Screenshot (474)
- Update User Success:
Screenshot (475)
- Update User Failed due to wrong ID:
Screenshot (476)
- Login User Authentication Successful:
Screenshot (477)
- Login User failed Invalid username or Password:
Screenshot (478)
- Login User failed user not found:
Screenshot (479)
- Delete User success:
Screenshot (480)
- Delete User failed due to User not found:
Screenshot (481)

## Dependencies (if any)

nothing to do

## Proof of Concept(s) (for new feature(s))

added features: Authenticaton, Get All Users, CRUD Users

## Other things that are not related to the JIRA ticket (if any)

nothing to do

### CODE REVIEW

File: azril_project/auth-user-api/controllers/user_controller.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+        }
+        return ctx.JSON(http.StatusBadRequest, echo.Map{"error": err.Error()})
+    }
+
+    if err := c.service.Register(req.Username, req.Email, req.Password1, req.Password2); err != nil {
+        return ctx.JSON(http.StatusBadRequest, echo.Map{"error": err.Error()})
+    }
+
+    // Mengembalikan respon sukses dengan format yang diinginkan
+    response := echo.Map{
```
Comment (albertsigiro) [inline azril_project/auth-user-api/controllers/user_controller.go]: kan saya sudah bilang buat struct response untuk nampung semua response, contoh nya bisa diliat di step_2, biar general responsenya, ini berlaku buat semua response. Tolong diperhatiin kalo ada masukan tuh ya

File: azril_project/auth-user-api/controllers/user_controller.go [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    service services.UserService
+}
+
+func NewUserController(service services.UserService) *UserController {
+    return &UserController{service}
+}
+
+// RegisterUser godoc
+func (c *UserController) RegisterUser(ctx echo.Context) error {
+    type RegisterRequest struct {
```
Comment (albertsigiro) [inline azril_project/auth-user-api/controllers/user_controller.go:23]: ini bisa kamu simpen di domains, dan tinggal panggil disini

File: azril_project/auth-user-api/controllers/user_controller.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    return ctx.JSON(http.StatusOK, echo.Map{
+        "message": "Users retrieved successfully",
+        "users":   users,
+        "code": "200",
+    })
+}
+
+// UpdateUser mengupdate informasi pengguna
+func (c *UserController) UpdateUser(ctx echo.Context) error {
+    type UpdateRequest struct {
```
Comment (albertsigiro) [inline azril_project/auth-user-api/controllers/user_controller.go:132]: ini bisa kamu simpen di domains, dan tinggal panggil disini

File: azril_project/auth-user-api/controllers/user_controller.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+            "email":     req.Email,
+            "password":  existingUser.Password, // diasumsikan password sudah di-hash di dalam database
+        },
+        "code": "200",
+    })
+}
+
+// DeleteUser godoc
+func (c *UserController) DeleteUser(ctx echo.Context) error {
+    type DeleteRequest struct {
```
Comment (albertsigiro) [inline azril_project/auth-user-api/controllers/user_controller.go:197]: ini bisa kamu simpen di domains, dan tinggal panggil disini

File: azril_project/auth-user-api/controllers/user_controller.go [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+
+    return ctx.JSON(http.StatusOK, echo.Map{
+        "message":         "User deleted successfully",
+        "code": "200",
+    })
+}
+
+// LoginUser godoc
+func (c *UserController) LoginUser(ctx echo.Context) error {
+    type LoginRequest struct {
```
Comment (albertsigiro) [inline azril_project/auth-user-api/controllers/user_controller.go:258]: ini bisa kamu simpen di domains, dan tinggal panggil disini

File: azril_project/auth-user-api/go.sum
Comment (albertsigiro) [inline azril_project/auth-user-api/go.sum:1]: kamu buat .gitignore buat file ini, biar tidak ikut ke push

File: azril_project/auth-user-api/controllers/user_controller.go (hunk 2 above)
Comment (arnoriel) [reply azril_project/auth-user-api/controllers/user_controller.go:23]: image

File: azril_project/auth-user-api/.gitignore
Comment (albertsigiro) [inline azril_project/auth-user-api/.gitignore:1]: .gitignore ini hapus aja, satu aja cukup

**Review by albertsigiro — APPROVED** [review]

**Pull request merged**